Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt

```diff
@@ -161,6 +180,20 @@ class InputOverlayDrawableJoystick(
                 y = touchY / maxY
 
                 setInnerBounds()
+                if (BooleanSetting.MAIN_OVERLAY_HAPTICS_JOYSTICK.boolean) {
+                    val radiusThreshold = gateRadius * 0.33
+                    val angularDistance = kotlin.math.abs(previousAngle - angle)
+                        .let { kotlin.math.min(it, Math.PI + Math.PI - it) }
+                    if (kotlin.math.abs(previousRadius - radius) > radiusThreshold
+                        || (radius > radiusThreshold &&
+                                (angularDistance >= HAPTICS_MAX_ANGLE || (radius == gateRadius &&
+                                        angularDistance * hapticsScale >= HAPTICS_MIN_ANGLE)))
```

Comment:
This provides feedback that you're moving the joystick, but it doesn't provide all that much feedback about where you currently are. What would you think of instead doing the following?

* If the joystick previously wasn't at the stick gate but now is at the stick gate, provide feedback.
* If the joystick previously was at the stick gate, and still is at the stick gate, and has hit one of the eight notches, and wasn't previously at that notch, provide feedback.

These actions provide tactile feedback on a real controller, and I think that they can provide useful information to the player about where the joystick currently is.